src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/YFlowUpdateService.java

```diff
@@ -0,0 +1,319 @@
+                YFlowUpdateContext context = YFlowUpdateContext.builder().subFlowId(flowId).build();
+                fsms.forEach((k, fsm) -> {
+                    if (fsm.getPendingSubFlows().contains(flowId)) {
+                        fsmExecutor.fire(fsm, Event.SUB_FLOW_ALLOCATED, context);
+                    }
+                });
+            }
+
+            @Override
+            public void onCompleted(String flowId) {
+                YFlowUpdateContext context = YFlowUpdateContext.builder().subFlowId(flowId).build();
+                fsms.forEach((k, fsm) -> {
+                    if (fsm.getPendingSubFlows().contains(flowId)) {
+                        fsmExecutor.fire(fsm, Event.SUB_FLOW_UPDATED, context);
+                    }
+                });
+            }
+
+            @Override
+            public void onFailed(String flowId, String errorReason, ErrorType errorType) {
+                YFlowUpdateContext context = YFlowUpdateContext.builder()
+                        .subFlowId(flowId)
+                        .error(errorReason)
+                        .errorType(errorType)
+                        .build();
+                fsms.forEach((k, fsm) -> {
+                    if (fsm.getPendingSubFlows().contains(flowId)) {
+                        fsmExecutor.fire(fsm, Event.SUB_FLOW_FAILED, context);
+                    }
+                });
+            }
+        });
+    }
+
+    /**
+     * Handles request for y-flow updating.
+     *
+     * @param key command identifier.
+     * @param request request data.
+     */
+    public void handleRequest(String key, CommandContext commandContext, YFlowRequest request)
+            throws DuplicateKeyException {
+        log.debug("Handling y-flow update request with key {} and flow ID: {}", key, request.getYFlowId());
+
+        if (fsms.containsKey(key)) {
+            throw new DuplicateKeyException(key, "There's another active FSM with the same key");
+        }
+
+        String eventKey = commandContext.getCorrelationId();
+        if (flowEventRepository.existsByTaskId(eventKey)) {
+            log.error("Attempt to reuse key {}, but there's a history record(s) for it.", eventKey);
+            return;
+        }
+
+        YFlowUpdateFsm fsm = fsmFactory.newInstance(commandContext, request.getYFlowId());
+        fsms.put(key, fsm);
+
+        YFlowUpdateContext context = YFlowUpdateContext.builder()
+                .targetFlow(request)
+                .build();
+        fsm.setTargetFlow(request);
+
+        fsmExecutor.fire(fsm, Event.NEXT, context);
+
+        removeIfFinished(fsm, key);
+    }
+
+    /**
+     * Handles request for y-flow patch updating.
+     *
+     * @param key command identifier.
+     * @param request request data.
+     */
+    public void handlePatchRequest(String key, CommandContext commandContext, YFlowPatchRequest request)
+            throws DuplicateKeyException {
+        YFlowRequest updateRequest = null;
```

Comment:
Can we name it something like `target` and `request` as `partialUpdate`?

Because names `request` and `updateRequest` do not provide enought data about who is who.

Reply:
Renamed to target.